docs/contributor/arch/007-log-pipeline-otlp-output.md

```diff
@@ -0,0 +1,52 @@
+# 7. LogPipeline OTLP Output
+
+Date: 2024-02-02
+
+## Status
+
+Accepted
+
+## Context
+
+Currently, `LogPipeline` is backed by Fluent Bit for log processing.
+However, we want to transition to OTel Collector to consolidate logging, tracing, and metrics capabilities of the Telemetry module.
+The challenge is to achieve this transition gradually without causing disruption.
+
+One potential solution is to introduce a new API version, allowing for the coexistence of a Fluent Bit backed `LogPipeline` and an OTel Collector backed `LogPipeline`. However, this approach comes with several drawbacks:
+
+* Two controllers are needed, each responsible for a different API version. It's possible to achieve it by appending a label to the `LogPipeline` resource during creation through a mutation webhook, and filtering by this label in the respective controllers.
+However, manual editing of the label remains possible.
+* The Kubernetes API server allows any client to request objects at any version, making it impossible to maintain clear separation between versions.
+
+An alternative approach involves introducing a new kind (e.g., `LogFlow`) or another API group (e.g., `logpipeline.flow.kyma-project.io`).
+However, this option is less favorable as it would require a substantial effort to migrate existing resources. We would also like to keep the old names as they accurately reflect the purpose of the resourcea.
```

Comment:
```suggestion
However, this option is less favorable as it would require a substantial effort to migrate existing resources. We would also like to keep the old names as they accurately reflect the purpose of the resource.
```